lib/addons/aws-loadbalancer-controller/index.ts

```diff
@@ -0,0 +1,96 @@
+import * as cdk from "@aws-cdk/core";
+import * as iam from "@aws-cdk/aws-iam";
+import { ClusterAddOn, ClusterInfo } from "../../stacks";
+import request from 'sync-request';
+
+/**
+ * Configuration options for AWS Load Balancer controller
+ */
+export interface AwsLoadBalancerControllerProps {
+    /**
+     * Namespace where controller will be installed
+     */
+    namespace?: string,
+    /**
+     * Version of the controller, i.e. v2.2.0 
+     */
+    version?: string,
+
+    /**
+     * Enable Shield (must be false for CN partition)
+     */
+    enableShield?: boolean,
+
+    /**
+     * Enable WAF (must be false for CN partition)
+     */
+    enableWaf: boolean,
+
+    /**
+     * Enable WAFV2 (must be false for CN partition)
+     */
+    enableWafv2?: boolean
+}
+
+/**
+ * Defaults options for load balancer controller.
+ */
+const AwsLoadBalancerControllerDefaults: AwsLoadBalancerControllerProps = {
+    namespace: 'kube-system',
+    version: 'v2.2.0',
+    enableShield: false,
+    enableWaf: false,
+    enableWafv2: false
+}
+
+const AWS_LOAD_BALANCER_CONTROLLER = 'aws-load-balancer-controller';
+
+export class AwsLoadBalancerControllerAddOn implements ClusterAddOn {
+
+    private options: AwsLoadBalancerControllerProps;
+
+    constructor(props?: AwsLoadBalancerControllerProps) {
+        this.options = { ...AwsLoadBalancerControllerDefaults, ...props ?? {}};
```

Comment:
Don't think a conditional is needed here?

Reply:
Yes, keep forgetting it is ts and a lot less restrictive. Will drop.